hack/extract.go

```diff
@@ -0,0 +1,253 @@
+	fset := token.NewFileSet()
+	r, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	//fmt.Printf("    Parsing %s\n", filename)
+	file, err := parser.ParseFile(fset, "", r, parser.ParseComments)
+	if err != nil {
+		return err
+	}
+
+	ast.Inspect(file, func(x ast.Node) bool {
+		fd, ok := x.(*ast.FuncDecl)
+		if !ok {
+			/*gd, ok := x.(*ast.GenDecl)
+			if !ok {
+				return true
+			}
+			for _, spec := range gd.Specs {
+				if vs, ok := spec.(*ast.ValueSpec); ok {
+					for _, v := range vs.Values {
+						if ue, ok := v.(*ast.UnaryExpr); ok {
+							fmt.Printf("%s: %s\n", ue.X, reflect.TypeOf(ue.X))
+						}
+					}
+				}
+			}*/
+			return true
+		}
+
+		for _, stmt := range fd.Body.List {
+			checkStmt(stmt, fd.Name.String(), f)
+		}
+		return true
+	})
+
+	return nil
+}
+
+func checkStmt(stmt ast.Stmt, parentFunc string, f string) {
+	// If this line is an expression, see if it's a function call
+	if t, ok := stmt.(*ast.ExprStmt); ok {
+		checkCallExpression(t, parentFunc, f)
+	}
+
+	// If this line is the beginning of an if statement, then check of the body of the block
+	if b, ok := stmt.(*ast.IfStmt); ok {
+		checkIfStmt(b, parentFunc, f)
+	}
+
+	// Same for loops
+	if b, ok := stmt.(*ast.ForStmt); ok {
+		for _, s := range b.Body.List {
+			checkStmt(s, parentFunc, f)
+		}
+	}
+}
+
+func checkIfStmt(stmt *ast.IfStmt, parentFunc string, f string) {
+	for _, s := range stmt.Body.List {
+		checkStmt(s, parentFunc, f)
+	}
+	if stmt.Else != nil {
+		// A straight else
+		if block, ok := stmt.Else.(*ast.BlockStmt); ok {
+			for _, s := range block.List {
+				checkStmt(s, parentFunc, f)
+			}
+		}
+
+		// An else if
+		if elseif, ok := stmt.Else.(*ast.IfStmt); ok {
+			checkIfStmt(elseif, parentFunc, f)
+		}
+
+	}
+}
+
+func checkCallExpression(t *ast.ExprStmt, parentFunc string, f string) {
+	if s, ok := t.X.(*ast.CallExpr); ok {
+		sf, ok := s.Fun.(*ast.SelectorExpr)
+		if !ok {
+			addFuncToList(parentFunc)
+			return
+		}
+		if f == sf.Sel.Name && len(s.Args) > 0 {
+			addFuncToList(parentFunc)
+			for _, arg := range s.Args {
+				// Find references to strings
+				if i, ok := arg.(*ast.Ident); ok {
+					if i.Obj != nil {
+						if as, ok := i.Obj.Decl.(*ast.AssignStmt); ok {
+							if rhs, ok := as.Rhs[0].(*ast.BasicLit); ok {
+								if addStringToList(rhs.Value) {
+									break
+								}
+							}
+						}
+					}
+				}
+				// Find string arguments
+				if argString, ok := arg.(*ast.BasicLit); ok {
+					if addStringToList(argString.Value) {
+						break
+					}
+				}
+			}
+		}
+	}
+}
+
+func addStringToList(s string) bool {
+	// Don't translate empty strings
+	if len(s) > 2 {
+		// Parse out quote marks
+		stringToTranslate := s[1 : len(s)-1]
+		// Don't translate integers
+		if _, err := strconv.Atoi(stringToTranslate); err != nil {
+			//Don't translate URLs
+			if u, err := url.Parse(stringToTranslate); err != nil || u.Scheme == "" || u.Host == "" {
```

Comment:
Make sure whatever example .go file you use tests this. It's a really interesting edge case.